Bali/src/main/java/OfferList.java

```diff
@@ -0,0 +1,117 @@
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Locale;
+import java.util.stream.Collectors;
+
+public class OfferList {
+    private List<Offer> offers;
+
+    public OfferList(){
+        offers = new ArrayList<>();
+    }
+
+    public Offer getOffer(String id){
+        for(Offer offer : offers){
+            if(offer.getId().equals(id)){
+                return offer;
+            }
+        }
+        System.out.println("Incorrect type of id!");
+        return null;
+    }
+
+    public static boolean validateOffer(Offer offer){
+        if(offer.getId() != null && offer.getId().length() != 0 &&
```

Comment:
You can make a step forward and add the Apache Commons Lang3 dependency to your project. This is "almost core" because is used in 90% of projects. Then, `offer.getDescription() != null && offer.getDescription().length()` turns into `StringUtils.isNotEmpty(offer.getDescription())`